### PR TITLE
stats-authors: toggle the "authors" module when there is only one to be displayed

### LINKED ISSUE #80534: Stats: Hide author box until site has multiple authors
Currently the author card is shown regardless of how many authors are there on the site. And if there is only one author for the site, it doesn't make sense to show the card as the information would be shown in page/post view already.

**Author Stats for site with only one author:**

Image

**Page/post views:**

Image

We could default the value to false if there's only one author unless turned on:

Image

### PR DESCRIPTION
**Pull request by macbre**
Resolves #80534

## Proposed Changes

* This PR toggles the stats "Authors" module when only a single author statistic can be rendered there.

## Testing Instructions

Find a site with just a single author active and a site with more active authors:

* http://calypso.localhost:3000/stats/day/farerskiekadry.pl
* http://calypso.localhost:3000/stats/week/`<your team's P2 domain>`?chartStart=2024-09-05&chartEnd=2024-12-03

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/features/modules/stats-authors/stats-authors.tsx [hunk 1]
```diff
@@ -45,6 +46,14 @@ const StatAuthors: React.FC< StatsDefaultModuleProps > = ( {
 		getSiteStatsNormalizedData( state, siteId, statType, query )
 	) as [ id: number, label: string ];
 
+	// toggle the module if it's going to show just a single author
+	const isSingleAuthor = data?.length === 1;
```
Comment (kangzj) [inline client/my-sites/stats/features/modules/stats-authors/stats-authors.tsx:50]: I'm afraid even if there is only one author in the response, but it wouldn't necessarily mean the site only has one author.

Comment (macbre) [reply client/my-sites/stats/features/modules/stats-authors/stats-authors.tsx:50]: What condition should I use instead?

Comment (jwebbdev) [reply client/my-sites/stats/features/modules/stats-authors/stats-authors.tsx:50]: There's a `useUsersQuery` that can take a `author_only: 1` option which may be able to get a list of authors on the site that you can check the size of.

Comment (kangzj) [reply client/my-sites/stats/features/modules/stats-authors/stats-authors.tsx:50]: I left a comment in the issue: 

https://github.com/Automattic/wp-calypso/issues/80534#issuecomment-2518462057

**Pull request closed without merging**